scene/gui/color_picker.cpp

```diff
@@ -422,14 +416,39 @@ ColorPicker::PickerShapeType ColorPicker::get_picker_shape() const {
 	return picker_type;
 }
 
+void ColorPicker::_add_preset_button(int p_size, const Color &p_color) {
+	ColorPresetButton *btn_preset = memnew(ColorPresetButton(p_color));
+	btn_preset->set_preset_color(p_color);
+	btn_preset->set_custom_minimum_size(Size2(p_size, p_size));
+	btn_preset->connect("gui_input", callable_mp(this, &ColorPicker::_preset_input), varray(p_color));
+	btn_preset->set_tooltip(vformat(RTR("Color: #%s\nLMB: Apply color\nRMB: Remove preset"), p_color.to_html(p_color.a < 1)));
+	preset_container->add_child(btn_preset);
+}
+
 void ColorPicker::add_preset(const Color &p_color) {
 	if (presets.find(p_color)) {
 		presets.move_to_back(presets.find(p_color));
+
+		// Find button to move to the end.
+		ColorPresetButton *btn_to_move = nullptr;
+		for (int i = 1; i < preset_container->get_child_count(); i++) {
+			ColorPresetButton *current_btn = Object::cast_to<ColorPresetButton>(preset_container->get_child(i));
+			if (current_btn && p_color == current_btn->get_preset_color()) {
+				btn_to_move = current_btn;
+				break;
+			}
+		}
+		if (btn_to_move) {
+			preset_container->move_child(btn_to_move, preset_container->get_child_count() - 1);
+		}
```

Comment:
You break the loop when you find button, so you could just move the button inside loop instead and then `btn_to_move` variable is not necessary. Same with removing few lines below.

Reply:
Done.